go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/1Password/shell-plugins
 
-go 1.18
+go 1.23.0
```

Comment:
`[non-blocking question]`

Wow, we haven't updated the Go version for so long. Can this have any potential problematic side-effects? 🤔 

I am aware of why this happened (other dependencies required this version as a minimum) and I am aware that the oldest supported version at the moment _is_ `1.23.0`. Therefore, I'm on board with this change, but I want to double-check we minimize the potential side-effects.